### PR TITLE
Allow default lights to be manually disabled.

### LINKED ISSUE #2376: Allow default lights to be disabled.
It would be nice to have the option of disabling default lights, without providing my own. For example, lighting and shadows can be baked into textures.

Proposed syntax:

```html
<a-scene light="enableDefaultLights: false">

  <!-- ... -->

</a-scene>
```

### PR DESCRIPTION
**Pull request by donmccurdy**
Fixes #2376.

### CODE REVIEW

File: tests/systems/light.test.js [hunk 1]
```diff
@@ -50,6 +50,20 @@ suite('light system', function () {
     el.appendChild(lightEl);
   });
 
+  test('it does not add default lights to scene if they are disabled', function (done) {
+    var el = this.el;
+    var lightEl = document.createElement('a-entity');
+    lightEl.setAttribute('light', {defaultLightsEnabled: false});
```
Comment (ngokevin) [inline tests/systems/light.test.js]: Shouldn't this be set on the scene and not an entity?

File: tests/systems/light.test.js [hunk 2]
```diff
@@ -50,6 +50,20 @@ suite('light system', function () {
...
+
+    assert.notOk(document.querySelectorAll('[light]').length);
+
+    lightEl.addEventListener('loaded', function () {
```
Comment (ngokevin) [inline tests/systems/light.test.js]: I'd wait until `scene.loaded`

File: tests/systems/light.test.js (hunk 1 above)
Comment (donmccurdy) [inline tests/systems/light.test.js]: Oops, too much copy/paste. Fixed.

File: tests/systems/light.test.js [hunk 3]
```diff
@@ -50,6 +50,22 @@ suite('light system', function () {
...
+  test('it does not add default lights to scene if they are disabled', function (done) {
+    var el = this.el;
+    var sceneEl = el.sceneEl;
+
+    // Systems cannot yet be updated via setAttribute().
+    sceneEl.systems.light.data.defaultLightsEnabled = false;
+
+    assert.notOk(document.querySelectorAll('[light]').length);
+
+    sceneEl.addEventListener('loaded', function () {
```
Comment (ngokevin) [inline tests/systems/light.test.js]: Looks like we don't need to listen if we're going to call the method directly.

File: tests/systems/light.test.js [hunk 4]
```diff
@@ -50,6 +50,22 @@ suite('light system', function () {
...
+      el.sceneEl.systems.light.setupDefaultLights();
+      assert.notOk(document.querySelectorAll('[' + DEFAULT_LIGHT_ATTR + ']').length);
```
Comment (ngokevin) [inline tests/systems/light.test.js]: I'd add a setTimeout since the method appends a DOM node.

File: tests/systems/light.test.js (hunk 3 above)
Comment (donmccurdy) [inline tests/systems/light.test.js]: Done.

File: tests/systems/light.test.js (hunk 4 above)
Comment (donmccurdy) [inline tests/systems/light.test.js]: Done.

**Pull request merged**